src/Beman/Optional26/tests/optional_constexpr.t.cpp

```diff
@@ -0,0 +1,1744 @@
+    EXPECT_TRUE(is_same);
+    EXPECT_TRUE(o1 == o2);
+
+    constexpr auto o3 = beman::optional26::make_optional<std::tuple<int, int, int, int>>(0, 1, 2, 3);
+    EXPECT_TRUE(std::get<0>(*o3) == 0);
+    EXPECT_TRUE(std::get<1>(*o3) == 1);
+    EXPECT_TRUE(std::get<2>(*o3) == 2);
+    EXPECT_TRUE(std::get<3>(*o3) == 3);
+
+    constexpr auto o5 = beman::optional26::make_optional<takes_init_and_variadic>({0, 1}, 2, 3);
+    EXPECT_TRUE(o5->v0 == 0);
+    EXPECT_TRUE(std::get<0>(o5->t) == 2);
+    EXPECT_TRUE(std::get<1>(o5->t) == 3);
+
+    constexpr static auto i  = 42;
+    constexpr auto o6 = beman::optional26::make_optional<const int&>(i);
+    static_assert(std::is_same<decltype(o6), const beman::optional26::optional<int>>::value);
+
+    EXPECT_TRUE((std::is_same<decltype(o6), const beman::optional26::optional<int>>::value));
+    EXPECT_TRUE(o6);
+    EXPECT_TRUE(*o6 == 42);
+}
+
+TEST(OptionalConstexprTest, Nullopt) {
+    constexpr beman::optional26::optional<int> o1 = beman::optional26::nullopt;
+    constexpr beman::optional26::optional<int> o2{beman::optional26::nullopt};
+    constexpr beman::optional26::optional<int> o3(beman::optional26::nullopt);
+    constexpr beman::optional26::optional<int> o4 = {beman::optional26::nullopt};
+
+    EXPECT_TRUE(!o1);
+    EXPECT_TRUE(!o2);
+    EXPECT_TRUE(!o3);
+    EXPECT_TRUE(!o4);
+
+    EXPECT_TRUE(!std::is_default_constructible<beman::optional26::nullopt_t>::value);
+}
+
+struct move_detector {
+    move_detector() = default;
+    move_detector(move_detector&& rhs) { rhs.been_moved = true; }
+    bool been_moved = false;
+};
+
+TEST(OptionalConstexprTest, Observers) {
+    constexpr beman::optional26::optional<int>       o1 = 42;
+    constexpr beman::optional26::optional<int>       o2;
+    constexpr beman::optional26::optional<int> o3 = 42;
+
+    EXPECT_TRUE(*o1 == 42);
+    EXPECT_TRUE(*o1 == o1.value());
+    EXPECT_TRUE(o2.value_or(42) == 42);
+    EXPECT_TRUE(o3.value() == 42);
+    constexpr auto success = std::is_same<decltype(o1.value()), const int&>::value;
+    EXPECT_TRUE(success);
+    constexpr auto success2 = std::is_same<decltype(o3.value()), const int&>::value;
+    EXPECT_TRUE(success2);
+    constexpr auto success3 = std::is_same<decltype(std::move(o1).value()), const int&>::value;
+    EXPECT_TRUE(success3);
+
+}
+
+namespace {
+class Point
+{
+    int x_;
+    int y_;
+  public:
+    constexpr Point() : x_(0), y_(0) {}
+    constexpr Point(int x, int y) : x_(x), y_(y) {}
+    auto operator<=>(const Point&) const = default;
+    bool operator==(const Point&) const = default;
+};
+
+}
+TEST(OptionalConstexprTest, RelationalOps) {
+    constexpr beman::optional26::optional<int> o1{4};
+    constexpr beman::optional26::optional<int> o2{42};
+    constexpr beman::optional26::optional<int> o3{};
+
+    //  SECTION("self simple")
+    {
+        {
+        constexpr auto b = !(o1 == o2);
+        EXPECT_TRUE(b);
+        }
```

Comment:
```C++
consteval auto constantify(auto expr){return expr;}
/// ...
EXPECT_TRUE(constantify(!(o1 == o2))
```
might be more readable, and a little less noise. At least less vertical space consumption for all these tests.